src/fault_injector/fault_types/fault.py

```diff
@@ -34,3 +34,26 @@ def inject_fault(component: Component):
         """
 
         raise NotImplementedError()
+
+    @abstractmethod
+    def resolve_fault(component: Component):
+        """resolves the previously injected fault
+
+        :param component: the component with the injected fault
+        :type component: Component
+        """
+
+        raise NotImplementedError()
+
+    def next_tick(self, tick: int):
+        """handle the next tick event accordingly
+
+        :param tick: the current simulation tick
+        :type tick: int
+        """
+        if tick == self.start_tick:
+            self.inject_fault(self.component)
+        elif tick == self.end_tick:
+            self.resolve_fault(self.component)
+
+        raise NotImplementedError
```

Comment:
```suggestion
```
You don't want to raise this/an error here. If you want subclasses to use this implementation (template-method style) remove it; if you want subclasses to override this functionality completely, remove the `if`-`elif`-block